src/utils/HighlighterCore/HighlighterCore.ts

```diff
@@ -0,0 +1,233 @@
+import removeAccents from 'remove-accents';
+
+export type HighlighterCoreAccuracy = 'exact' | 'partial';
+
+type Substring = {
+    highlighted: boolean;
+    substring: string;
+};
+
+type SubstringIndexes = {
+    start: number;
+    end: number;
+};
+
+interface RegExpBuilder {
+    searchString: string | RegExp;
+    accuracy?: HighlighterCoreAccuracy;
+    ignoreCase?: boolean;
+}
+
+interface StringNormalizer {
+    string: string;
+    useEscape?: boolean;
+    ignoreDiacritics?: boolean;
+}
+
+export interface HighlighterCoreOptions {
+    /** Array of words or RegExp that is used for search */
+    searchTerms: string[] | RegExp;
+    /**
+     * Define the accuracy of search
+     * Supported options:
+     * partially - when searching for "lor" only "lor" inside "lorem" will be marked
+     * exactly - When searching for "lor" only those exact words with a word boundary will be marked.
+     * regexp - When searching for "lor" only those exact words with a word boundary will be marked.
+     */
+    accuracy?: HighlighterCoreAccuracy;
+    /** If enabled the diacritic characters should be matched */
+    ignoreDiacritics?: boolean;
+    /** Ignore case sensitivity */
+    ignoreCase?: boolean;
+}
+
+export class HighlighterCore {
+    protected accuracy: HighlighterCoreAccuracy;
+
+    protected regexp: RegExp | null;
+
+    protected ignoreDiacritics: boolean;
+
+    protected ignoreCase: boolean;
+
+    static escapeStr(string: string) {
+        return string.replace(/[-[\]{}():*+?.,\\^$|#\s]/g, '\\$&');
+    }
+
+    static removeDiacritics(string: string) {
+        return removeAccents(string);
+    }
+
+    static buildRegExp({ searchString, accuracy, ignoreCase }: RegExpBuilder) {
+        let source: string | RegExp = '';
+        let flags = 'gm';
+
+        switch (accuracy) {
+            case 'partial':
+                source = `(${searchString})`;
+                break;
+            case 'exact':
```

Comment:
using of this `case` can be omitted since it falls to `default` keyword

Reply:
I would like to keep  `exact` because we can explicitly see which code applies for each `accuracy`. We can skip default, but it is not a good practice :)